code/controllers/ExtensibleSearchAPI.php

```diff
@@ -93,22 +94,43 @@ public function toggleSuggestionApproved($request) {
 	 *	@URLparameter term <{SEARCH_TERM}> string
 	 *	@return JSON
 	 */
-
 	public function getSuggestions($request) {
 
-		if(Config::inst()->get('ExtensibleSearchSuggestion', 'enable_suggestions') && ($suggestions = $this->service->getSuggestions($request->getVar('term'), $request->getVar('page')))) {
+		if(Config::inst()->get('ExtensibleSearchSuggestion', 'enable_suggestions')) {
+			$suggestions = $this->service->getSuggestions($request->getVar('Search'), $request->getVar('page'));
 
 			// Return the search suggestions as JSON.
 
 			$this->getResponse()->addHeader('Content-Type', 'application/json');
 
-			// JSON_PRETTY_PRINT.
+			// 128 = JSON_PRETTY_PRINT, const added in 5.4
+			return json_encode($suggestions. 128);
+		} else {
 
-			return json_encode($suggestions, 128);
-		}
-		else {
 			return $this->httpError(404);
 		}
 	}
 
+	/**
+	 *	Retrieve search typeahead suggestions
+	 *
+	 *	@URLparameter term <{SEARCH_TERM}> string
+	 *	@return JSON
+	 */
+	public function getTypeahead($request) {
+
+		if(Config::inst()->get('ExtensibleSearchSuggestion', 'enable_typeahead')	) {
+
+			$typeahead = $this->service->getTypeahead($request->getVars(), $request->getVar('page'));
```

Comment:
This doesn't actually look to be used if that's the case (from the service definition below)? Considering this is a service, I would still rather the individual GET variables passed through where required, as it may be called elsewhere without an array of request GET variables available. If you do however still require the request object, this should be done outside the service (which may be the get search engine results in your case).